databases/week1/homework.sql

```diff
@@ -0,0 +1,48 @@
+-- Find out how many tasks are in the task table
+SELECT COUNT(id)
+FROM task;
+-- Find out how many tasks in the task table do not have a valid due date
+SELECT COUNT(id)
+FROM task
+WHERE due_date IS NULL;
+-- Find all the tasks that are marked as done
+SELECT *
+FROM task
+WHERE status_id=(SELECT id FROM status WHERE name='done'
+ );
+ -- Find all the tasks that are not marked as done
+SELECT *
+FROM task
+WHERE status_id IN (SELECT id FROM status WHERE name !='done'
+ );
+ -- Get all the tasks, sorted with the most recently created first
+ SELECT *
+ FROM task
+ ORDER BY created DESC;
+ -- Get the single most recently created task
+ SELECT *
+ FROM task
+ ORDER BY created DESC
+ LIMIT 1;
+ -- Get the title and due date of all tasks where the title or description contains database
+ SELECT title, due_date
+ FROM task
+ WHERE title LIKE '%database%' OR description LIKE '%database%';
+ -- Get the title and status (as text) of all tasks
+ SELECT title, name
```

Comment:
it is a good practice to use the table alias when you select fields. 
- It is more readable 
- it will prevent an error in the case when both tables have the same field 
```suggestion
 SELECT t.title, s.name
```